fastfood/food.py

```diff
@@ -42,33 +41,23 @@ def build_cookbook(build_config, templatepack, cookbooks_home,
     with open(build_config) as cfg:
         cfg = json.load(cfg)
 
+    cookbook_name = cfg['name']
     cookbook = None
-    if cookbook_path:
-        try:
-            cookbook = book.CookBook(cookbook_path)
-        # change this to a more specific error
-        except ValueError:
-            # create a new one please
-            pass
+    cookbook_path = os.path.join(cookbooks_home, cookbook_name)
 
     written_files = []
     if not cookbook:
-        cookbook_name = cfg['name']
-        files, cookbook = create_new_cookbook(
-            cookbook_name, templatepack, cookbooks_home, force=force)
-        written_files.extend(files)
+        cookbook, cookbook_path = create_new_cookbook(
+            cookbook_name, cookbooks_home, force=force)
 
     for stencil in cfg['stencils']:
         stencil_set = stencil.pop('stencil_set')
         # items left un-popped are **options
-        files, updated_cookbook = update_cookbook(
-            cookbook, templatepack, stencil_set, **stencil)
+        files, updated_cookbook = process_stencil(cookbook_name, cookbook_path, templatepack, stencil_set, **stencil)
         written_files.extend(files)
-        assert updated_cookbook is cookbook
     return written_files, cookbook
 
-
-def update_cookbook(cookbook, templatepack, stencilset_name, **options):
+def process_stencil(cookbook_name, cookbook_path, templatepack, stencilset_name, **options):
```

Comment:
Do you have the `Cookbook` instance on hand when calling this? If so, could you change the signature to just require a `cookbook` argument and use `cookbook.name` and `cookbook.path` ?